services/python/python-version-from-toml.tester.js

```diff
@@ -0,0 +1,26 @@
+import Joi from 'joi'
+import { createServiceTester } from '../tester.js'
+export const t = await createServiceTester()
+
+const isCommaSeparatedPythonVersions = Joi.string().regex(
+  // This should test for PEP440
+  // Accepted values are one or more Version specifiers as defined at https://peps.python.org/pep-0440/#version-specifiers
+  // Some strings might include spaces, those are valid, values are comma seperated
+  // Versions should fit the version scheme https://peps.python.org/pep-0440/#version-scheme
+  // This is based on the example in PEP440 at https://peps.python.org/pep-0440/#appendix-b-parsing-version-strings-with-regular-expressions
+  /^\s*(?:(?:===|!=|<=|>=|<|>)\s*)?((?:(?:\d+!)?(?:\d+(?:\.\d+)*))(?:(?:[abc]|rc)\d+)?(?:\.post\d+)?(?:\.dev\d+)?)(?:\s*,\s*(?:(?:===|!=|<=|>=|<|>)\s*)?((?:(?:\d+!)?(?:\d+(?:\.\d+)*))(?:(?:[abc]|rc)\d+)?(?:\.post\d+)?(?:\.dev\d+)?))*\s*$/,
+)
```

Comment:
Rather than copy/pasing this monster regex (I hope you didn't write this from scratch! :smile: ), I'd suggest we use `Joi.custom()` https://joi.dev/api/#anycustommethod-description here to make a validator using the `validRange()` function from `@renovate/pep440` (which is a package we already use).

```js
import pep440 from '@renovate/pep440'
pep440.validRange(">=3.6, <4.0") // true
```

Reply:
Should be fixed with https://github.com/badges/shields/pull/9516/commits/b3de30f1859274749dc0e513be77ecd2b63ddfcd
I still have issue testing so i feel like mountain climbing where i try to get it first try, lets see the  CI results.